src/middlewared/middlewared/plugins/netdata.py

```diff
@@ -0,0 +1,241 @@
+                if not data.get(key):
+                    verrors.add(
+                        f'netdata_update.{key}',
+                        f'{key} is required with stream mode as MASTER'
+                    )
+
+        verrors.check()
+
+        for alarm in valid_alarms:
+            # Let's add alarms to our db if they aren't already there
+            if alarm not in data['alarms']:
+                data['alarms'][alarm] = True
+
+        for alarm in update_alarms:
+            # These are valid alarms
+            data['alarms'][alarm] = update_alarms[alarm]
+
+        return data
+
+    @accepts(
+        Dict(
+            'netdata_update',
+            Str('additional_params'),
+            Dict(
+                'alarms',
+                additional_attrs=True
+            ),
+            List('allow_from', items=[Str('pattern')]),
+            Str('api_key', validators=[UUID()]),
+            List('bind', items=[Str('bind_ip')]),
+            Int('port', validators=[Port()]),
+            List('destination', items=[Str('destination')]),
+            Int('history'),
+            Int('http_port_listen_backlog'),
+            Str('stream_mode', enum=['NONE', 'MASTER', 'SLAVE']),
+            Int('update_every')
+        )
+    )
+    async def do_update(self, data):
+        """
+        Update Netdata Service Configuration
+
+        `alarms` is a dictionary where user specifies a key,value pair with key being alarm name and value is a boolean
+        value which represents should the alarm be enabled or not. Middlewared supports interacting (changing) alarms
+        in /usr/local/etc/netdata/health.d/ directory.
+
+        `allow_from` is used when netdata service is expected to be used as a master. It defaults to "['*']". This field
+        expects a list of Netdata patterns which Netdata will use to set restrictions on incoming connections from slave
+        accordingly.
+
+        `api_key` is a valid UUID which can be generated in command line by typing uuidgen.
+
+        `destination` is used when netdata service is expected to be used as a slave. Destination is a list of potential
+        destinations to which netdata should stream metrics. We expect the format to be IP:PORT ( port is optional ).
+        The first working destination is used by Netdata service.
+
+        `history` is the number of entries the netdata daemon will by default keep in memory for each chart dimension.
+        It defaults to 86400.
+
+        .. examples(websocket)::
+
+          Update Netdata Service Configuration
+
+            :::javascript
+            {
+                "id": "6841f242-840a-11e6-a437-00e04d680384",
+                "msg": "method",
+                "method": "netdata.update",
+                "params": [{
+                    "history": 80000,
+                    "alarms": {
+                        "used_swap": "true",
+                        "ram_in_swap": "true"
```

Comment:
Is this right? "true" as string? Are there any other values accepted other than a boolean?

Reply:
For alarms no. We only allow the alarm to be enabled/disabled. Thus a boolean value

Reply:
Actually I have to omit the "" from true if I am not wrong